libethcore/Miner.h

```diff
@@ -253,11 +253,16 @@ class Miner: public Worker
 		kick_miner();
 	}
 
-	uint64_t hashCount() const { return m_hashCount.load(std::memory_order_relaxed); }
-
-	void resetHashCount() { m_hashCount.store(0, std::memory_order_relaxed); }
-
-	unsigned Index() { return index; };
+    uint64_t RetrieveAndClearHashCount()
+    {
+        uint64_t expected;
+        do
+            expected = m_hashCount.load(std::memory_order_relaxed);
```

Comment:
This load should be done only once because the compare_exchange_weak will load it on failure.
It should look like this:

```
uint64_t expected = m_hashCount.load(std::memory_order_relaxed);
while (!m_hashCount.compare_exchange_weak(expected, 0, std::memory_order_relaxed)) {}
return expected;
```